include/rocksdb/table.h

```diff
@@ -79,19 +127,45 @@ struct BlockBasedTableOptions {
   // than data blocks.
   bool cache_index_and_filter_blocks_with_high_priority = true;
 
+  // DEPRECATED: This option will be removed in a future version. For now, this
+  // option still takes effect on relevant `CachePinningOptions` members that
+  // are set to `PinningTier::kFallback`. In particular, the overrides from
+  // setting this option (and the changes users should make to migrate away from
+  // setting this option) are:
+  //
+  // - `metablock_partition_pinning == PinningTier::kFallback` ->
+  //   `metablock_partition_pinning == PinningTier::kMaybeFlushed`
+  // - `unpartitioned_metablock_pinning == PinningTier::kFallback` ->
+  //   `unpartitioned_metablock_pinning == PinningTier::kMaybeFlushed`
+  //
```

Comment:
I am not sure I understand the logic here for how the options are used.  It seems like if the old option was false (the default) these values should be whatever their default is (which I assume is "off"?).  If the old option is true and the new options are their default, then set them to the expected "default" values.  Does that make it cleaner?

Reply:
Yes, I tried rewriting it.